index.html

```diff
@@ -63,17 +66,60 @@ <h1>liczba wyborcow w danych okregach wyborczych</h1>
                     var selectedDistrict = rows[selectedOption].split(';')[0];
 
                     var districtRows = rows.filter(function (row) {
-                    return row.startsWith(selectedDistrict + ';');
+                        return row.startsWith(selectedDistrict + ';');
                     });
 
                     var squaredDistance = 0;
                     for (var i = 0; i < districtRows.length; i++) {
-                    var rowData = districtRows[i].split(';');
-                    var votersCount = parseFloat(rowData[4].replace(',', '.')) || 0;
-                    squaredDistance += Math.pow(votersCount, 2);
+                        var rowData = districtRows[i].split(';');
+                        var votersCount = parseFloat(rowData[4].replace(',', '.')) || 0;
+                        squaredDistance += Math.pow(votersCount, 2);
                     }
 
                     document.getElementById('kwadratodleglosci').innerHTML = 'Kwadrat odległości dla Okręgu Wyborczego ' + selectedDistrict + ': ' + squaredDistance;
+
+                    findClosestDistricts(selectedDistrict, squaredDistance);
+                };
+
+                reader.readAsText(loadedFile);
+            } else {
+                alert('Proszę najpierw załadować plik CSV i wybrać okręg.');
+            }
+        }
+
+        function findClosestDistricts(selectedDistrict, currentSquaredDistance) {
+            var selectedOption = document.getElementById('select').value;
+            if (loadedFile && selectedOption !== "") {
+                var reader = new FileReader();
+                reader.onload = function (e) {
+                    var csvData = e.target.result;
+                    var rows = csvData.split('\n');
+
+                    var distances = [];
+
+                    for (var i = 0; i < addedDistricts.length; i++) {
+                        var districtNumber = addedDistricts[i];
+                        if (districtNumber !== selectedDistrict) {
+                            var districtRows = rows.filter(function (row) {
+                                return row.startsWith(districtNumber + ';');
+                            });
+
+                            var squaredDistance = 0;
+                            for (var j = 0; j < districtRows.length; j++) {
+                                var rowData = districtRows[j].split(';');
+                                var votersCount = parseFloat(rowData[4].replace(',', '.')) || 0;
+                                squaredDistance += Math.pow(votersCount, 2);

```

Comment:
I see you are calculating the `squaredDistance` here.

Remember the formula we arrived at to get the distance between two districts:

![image](https://github.com/Kubast06/Praktyki/assets/889383/9327043c-2688-452f-af9d-0e82ad43f152)

What you implement here is sort of like that formula, but it is missing a critical piece: comparing **two** districts, not one.

In the `x^2 = a^2 + b^2 + ...` formula, the components `a`, `b`, are the **differences** between the scores of some committee (e.g. `a` for PO, `b` for PIS) of 2 different districts.

Notice that in the screenshot on the graph, the distance is between 2 points. What you are calculating here, is the squared distance of some point from the origin of the graph (point `(0, 0)`).

What you need to do, is the **subtract** the number of voters from the district, from the number of voters for that committee in the selected district. Then, you square that difference.

It could look something like:

```ts
squaredDistance += Math.pow(votersCount - selectedOptionVotersForThisCommittee, 2);
```

For example, when `votersCount` is the number of votes for PIS in the district 2, and the user selected the district 5, then you need to find the number of voters for PIS in district 5, and subtract the 2 numbers, and then square them. Only then do you get a distance between **two** point.